### PR TITLE
Proxy support for file_get_contents

### PR DESCRIPTION
**Pull request by symfonyaml**
Proxy support using:
`export HTTPS_PROXY=proxy:port`

### CODE REVIEW

**Comment by ppassmannpriv** [comment]
This is great - should be merged into master to be frank. I'm waiting for this feature :x

**Comment by cwreden** [comment]
Hi, when is this feature merged?

File: SensioLabs/Security/Crawler.php [hunk 1]
```diff
@@ -97,6 +97,19 @@ private function doCheck($lock, $format = 'json', array $contextualHeaders = [])
...
 
+        $proxy = getenv('HTTPS_PROXY');
+        if ($proxy) {
+            $parts = parse_url($proxy);
+            if (!is_array($parts) || (!isset($parts['path']) && !isset($parts['host']))) {
+                throw new RuntimeException(sprintf('Failed to parse HTTPS_PROXY "%s".', $proxy));
+            }
+            $opts['http']['proxy'] = sprintf(
+                '%s:%d',
+                isset($parts['host']) ? $parts['host'] : $parts['path'],
```
Comment (maxhelias) [inline SensioLabs/Security/Crawler.php]:
```suggestion
                $parts['host'] ?? $parts['path'],
```

File: SensioLabs/Security/Crawler.php [hunk 2]
```diff
@@ -97,6 +97,19 @@ private function doCheck($lock, $format = 'json', array $contextualHeaders = [])
...
+                isset($parts['port']) ? $parts['port'] : 80
```
Comment (maxhelias) [inline SensioLabs/Security/Crawler.php]:
```suggestion
                $parts['port'] ?? 80
```

File: SensioLabs/Security/Crawler.php [hunk 3]
```diff
@@ -97,6 +97,19 @@ private function doCheck($lock, $format = 'json', array $contextualHeaders = [])
             ],
         ];
 
+        $proxy = getenv('HTTPS_PROXY');
+        if ($proxy) {
+            $parts = parse_url($proxy);
+            if (!is_array($parts) || (!isset($parts['path']) && !isset($parts['host']))) {
```
Comment (maxhelias) [inline SensioLabs/Security/Crawler.php]:
```suggestion
            if (!is_array($parts) || !isset($parts['path'], $parts['host'])) {
```

File: SensioLabs/Security/Crawler.php [hunk 4]
```diff
@@ -97,6 +97,19 @@ private function doCheck($lock, $format = 'json', array $contextualHeaders = [])
             ],
         ];
 
+        $proxy = getenv('HTTPS_PROXY');
+        if ($proxy) {
```
Comment (maxhelias) [inline SensioLabs/Security/Crawler.php]:
```suggestion
        if ($proxy = getenv('HTTPS_PROXY')) {
```

File: SensioLabs/Security/Crawler.php (hunk 1 above)
Comment (BackEndTea) [reply SensioLabs/Security/Crawler.php]: `??` wasnt'added untill php 7.0, and this project supports php versions lower than that, so it can not be used.

**Comment by nicolas-grekas** [comment]
Same comment as #141: symfony/symfony#30413 will solve this issue in a much cleaner way, I'd suggest waiting for it to be merged.

File: SensioLabs/Security/Crawler.php (hunk 1 above)
Comment (maxhelias) [reply SensioLabs/Security/Crawler.php]: Oops as much for me 😅

**Pull request closed without merging**